src/test/scala/com/feedzai/cosytest/testng/IntegrationSpec.java

```diff
@@ -0,0 +1,76 @@
+package com.feedzai.cosytest.testng;
+
+import com.feedzai.cosytest.DockerComposeJavaManager;
+import com.feedzai.cosytest.DockerComposeJavaSetup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.ITestResult;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.HashMap;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+
+
+public class IntegrationSpec {
+
+    private static Logger logger = LoggerFactory.getLogger(IntegrationSpec.class);
+
+    private DockerComposeJavaSetup dockerSetup;
+
+    private DockerComposeJavaManager containerManager;
+
+    @BeforeClass
+    public void init() {
+
+        dockerSetup = new DockerComposeJavaSetup(
+                "testngtest",
+                Collections.singletonList(Paths.get("src", "test", "resources", "docker-compose-junit.yml")),
+                Paths.get("").toAbsolutePath(),
+                new HashMap<>()
+        );
+
+        containerManager = DockerComposeJavaManager
+                .builder(dockerSetup)
+                .build();
+
+        containerManager.launchContainers();
+    }
+
+    @AfterMethod
+    public void checkTestResult(ITestResult result) {
+        if (result.getStatus() == ITestResult.FAILURE) {
+            containerManager.setTestFailed(true);
+        }
+    }
+
+    @AfterClass
+    public void tearDown() {
+        containerManager.tearDownContainers();
+    }
+
+    @Test
+    public void fetchServices() {
+        assertThat(
+                dockerSetup.getServices(),
+                containsInAnyOrder("container1", "container2", "container3")
+        );
+    }
+
+    @Test
+    public void validateMappedPorts() {
+        assertThat(dockerSetup.getServiceMappedPort("container1", 80), contains("8084"));
+        assertThat(dockerSetup.getServiceMappedPort("container2", 80), contains("8085"));
+        assertThat(dockerSetup.getServiceMappedPort("container3", 80), contains("8086"));
+    }
+
+}
+
+
```

Comment:
Can't we just create a class that would implement the @BeforeClass, @AfterMethod and @AfterClass and then could be extended?
And there it could be added the start up and teardown instead of having a new DockerComposeJava wrapper.

Reply:
Could you explain please, I don't think I follow.
Specifically what do you mean by:
`There it could be added the start up and teardown instead of having a new DockerComposeJava wrapper.`

Reply:
What I was trying to say is that you can create an abstract class which could have the method annotated with @BeforeClass and in that method you could have the same that is being done in the launchContainers.
The same could be done for @AfterClass.

Reply:
IIUC you are saying passing the init method to an abstract class that is part of the cosy test api for testng? If I do that, then how can users of the testng API, modify the DockerComposeJavaSetup object to their purposes and possibly (as is my case) run some arbitrary piece of code by constructing and running the DockerComposeJavaSetup object ?